### DISCUSSION TITLE
Use one single runner instance across the duration of an Action/Workflow?

### DISCUSSION THREAD

**Discussion by gbakeman** [community/community#52339] [Actions] [answered]
### Select Topic Area

Question

### Body

I'm getting further into CI/CD and loving it so far. My project is configured to use a GitHub-hosted runner which has been working fine for the most part. I'm experimenting with splitting up workflows into reusable units, which means I'm splitting up my automatic, purpose-built workflows into multiple jobs which either invoke the reusable workflows, or run specific tasks I've written for any given need (debug/release/translations, etc.)

I'd like to be able to preserve a single runner environment across the jobs of one workflow run, but I'm having trouble figuring out how to do this. Is this possible?

Edit: Unless I'm misunderstanding, then [according to the docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#using-a-github-hosted-runner) it does look like runners are restricted to individual jobs. It would be very convenient if a series of jobs could be run on one runner instance, though.

### See Also
- _StackOverflow_ [Github actions: Run multiple jobs in a single runner or share workspace between jobs](https://stackoverflow.com/questions/71579731/github-actions-run-multiple-jobs-in-a-single-runner-or-share-workspace-between)
- _StackOverflow_ [In a github actions workflow, is there a way to have multiple jobs reuse the same setup?](https://stackoverflow.com/questions/65242830/in-a-github-actions-workflow-is-there-a-way-to-have-multiple-jobs-reuse-the-sam)
- _StackOverflow_ [Github actions share workspace/artifacts between jobs?](https://stackoverflow.com/questions/57498605/github-actions-share-workspace-artifacts-between-jobs)
- https://github.com/actions/runner/issues/646

**Comment by airtower-luna** [comment, marked as answer]
> Edit: Unless I'm misunderstanding, then [according to the docs](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#using-a-github-hosted-runner) it does look like runners are restricted to individual jobs. It would be very convenient if a series of jobs could be run on one runner instance, though.

That is correct.

You might want to consider using [composite actions](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) for your building blocks, instead of reusable workflows. That way you could have one job that calls the actions you want (`uses`), with later actions having access to the environment created by the previous ones.

**Reply by gbakeman** [reply]
That's the ticket - I've just started taking a deeper dive into composite actions and it looks like that's exactly what I need. Thank you!
